redhat-developer/devspaces
# chore: add meta.yaml for ansible sample

### What does this PR do?
Registers **ansible-demo** sample. The sample was forked from https://github.com/kyetter/ansible-devspace-demo and it's located in https://github.com/devspaces-samples/ansible-devspaces-demo.

This sample has **Community** label and uses **quay.io/devspaces/ansible-creator-ee** as a tools image. The default branch that uses for the next DS version uses **latest** tag and branches with specific DS version use appropriate tag (e.g. devspaces-**3.8**-rhel-8 will use quay.io/devspaces/ansible-creator-ee:**3.8** image)
<details>
  <summary>The whole devfile.yaml is</summary>

 ```yaml
schemaVersion: 2.2.0
metadata:
  name: ansible-demo
components:
  - name: tooling-container
    container:
      image: quay.io/mloriedo/ansible-creator-ee:latest
      memoryRequest: 256M
      memoryLimit: 6Gi
      cpuRequest: 250m
      cpuLimit: 2000m
      args: ['tail', '-f', '/dev/null']
commands:
  - id: oc-install
    exec:
      commandLine: '/usr/local/bin/ansible-playbook ${PROJECT_SOURCE}/playbooks/install_oc.yml'
      workingDir: ${PROJECT_SOURCE}
      group:
        kind: build
        isDefault: true
      component: tooling-container
  - id: molecule-test
    exec:
      label: "Molecule: Run Scenario for Backup Role"
      commandLine: 'source $HOME/.bashrc && molecule test'
      workingDir: ${PROJECTS_ROOT}/ansible-devspace-demo/roles/backup_file
      group:
        kind: run
        isDefault: true
      component: tooling-container
  - id: molecule-verify
    exec:
      label: "Molecule: Validate Backup File Role"
      commandLine: 'source $HOME/.bashrc && molecule verify'
      workingDir: ${PROJECTS_ROOT}/ansible-devspace-demo/roles/backup_file
      group:
        kind: run
        isDefault: true
      component: tooling-container
events:
  postStart:
    - "oc-install"
```
</details>

<details>
  <summary>Dockerfile is located in the sample's repo and its content is</summary>

```dockerfile
FROM quay.io/ansible/creator-ee

ENV HOME /home/runner

# nodejs 16 + VSCODE_NODEJS_RUNTIME_DIR are required on ubi9 based images
# until we fix https://github.com/eclipse/che/issues/21778
# When fixed, we won't need this Dockerfile anymore.
# c.f. https://github.com/che-incubator/che-code/pull/120
RUN \
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh | bash && \
export NVM_DIR="$HOME/.nvm" && \
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" && \
nvm install 16.20.0
ENV VSCODE_NODEJS_RUNTIME_DIR="$HOME/.nvm/versions/node/v16.20.0/bin/"
```

</details>

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-4359

### How to test?
1. Start a workspace using a [factory link for dev-sandbox](https://devspaces.apps.sandbox-m4.g2pi.p1.openshiftapps.com/dashboard/#/https://github.com/devspaces-samples/ansible-devspaces-demo)
2. When the workspace is ready, open **hello_world.yml** file. _Ansible-lint_ should validate the content and show an error.
3. Try to type something using _autocomplete_ feature 
4. Run commands from the devfile
-  **Molecule: Run Scenario for Backup Role** (will run whole scenario for backup role, the source code of this scenario is located in /projects/ansible-devspace-demo/roles/backup_file/molecule). As the result of this command you should see 4 new pods in the user namespace that were installed by ansible.
- **Molecule: Validate Backup File Role** (will run only **verify** molecule scenario on the instances)


### Screenshots

![screenshot-devspaces apps ocp412-aws crw-qe com-2023 06 07-15_34_21](https://github.com/redhat-developer/devspaces/assets/1271546/ec5cc110-1732-42b6-a1c9-48187d999abc)

![screenshot-devspaces apps ocp412-aws crw-qe com-2023 06 07-15_48_55](https://github.com/redhat-developer/devspaces/assets/1271546/2a15c4f0-bd53-408b-9efd-5a0a9d3161e9)





#### Release Notes

Add an Ansible sample into the devfile registry. The sample uses community image in the main component.
